internetarchive/cli/ia_upload.py

```diff
@@ -260,8 +260,12 @@ def main(argv, session):
     # Bulk upload using spreadsheet.
     else:
         # Use the same session for each upload request.
-        with io.open(args['--spreadsheet'], 'rU', newline='', encoding='utf-8') as csvfp:
-            spreadsheet = csv.DictReader(csvfp)
+
+        with io.open(args['--spreadsheet'], 'rU', newline='', encoding='utf-8') as fp:
+            if is_json(args['--spreadsheet']):
+                spreadsheet = json.load(fp)
+            else:
+                spreadsheet = csv.DictReader(fp)
```

Comment:
It is strange to call a `dict` a `spreadsheet`.
Does the `for row in spreadsheet:` logic below really work for all .json files?

Reply:
The `csv.DictReader` is an iterator producing a `dict` for each row in the CSV file. You only get `dict`s in the loop below.
All that means is that the JSON file must have a similar structure, i.e. an array of objects:

```json
[
	{"identifier": "item", "file": "./foobar"},
	{"file": "./foobaz"}
]
```

This needs test coverage though and an example in the documentation.